app/graphql/mutations/create_submission_mutation.rb

```diff
@@ -1,50 +1,41 @@
 # frozen_string_literal: true
 
 module Mutations
-  module CreateSubmissionMutation
-    Definition =
-      GraphQL::Relay::Mutation.define do
-        name 'CreateSubmissionMutation'
+  class CreateSubmissionMutation < Mutations::BaseMutation
+    argument :artist_id, String, required: true
 
-        input_field :additionalInfo, types.String
-        input_field :artistID, !types.String
-        input_field :authenticityCertificate, types.Boolean
-        input_field :category, Types::CategoryType
-        input_field :currency, types.String
-        input_field :depth, types.String
-        input_field :dimensionsMetric, types.String
-        input_field :edition, types.Boolean
-        input_field :editionNumber, types.String
-        input_field :editionSize, types.Int
-        input_field :height, types.String
-        input_field :locationCity, types.String
-        input_field :locationCountry, types.String
-        input_field :locationState, types.String
-        input_field :medium, types.String
-        input_field :minimumPriceDollars, types.Int
-        input_field :provenance, types.String
-        input_field :signature, types.Boolean
-        input_field :state, Types::StateType
-        input_field :title, types.String
-        input_field :width, types.String
-        input_field :year, types.String
+    argument :additional_info, String, required: false
+    argument :authenticity_certificate, Boolean, required: false
+    argument :category, Types::CategoryType, required: false
+    argument :currency, String, required: false
+    argument :depth, String, required: false
+    argument :dimensions_metric, String, required: false
+    argument :edition, Boolean, required: false
+    argument :edition_number, String, required: false
+    argument :edition_size, Integer, required: false
+    argument :height, String, required: false
+    argument :location_city, String, required: false
+    argument :location_country, String, required: false
+    argument :location_state, String, required: false
+    argument :medium, String, required: false
+    argument :minimum_price_dollars, Integer, required: false
+    argument :provenance, String, required: false
+    argument :signature, Boolean, required: false
+    argument :state, Types::StateType, required: false
+    argument :title, String, required: false
+    argument :width, String, required: false
+    argument :year, String, required: false
 
-        return_field :consignmentSubmission, Types::SubmissionType
-      end
+    field :consignment_submission, Types::SubmissionType, null: true
 
-    def self.resolve(_obj, args, context)
-      params = args.to_h['input'].except('clientMutationId')
-      client_mutation_id = args.to_h['input']['clientMutationId']
+    def resolve(arguments)
+      resolve_options = {
+        arguments: arguments, context: context, object: object
+      }
+      resolver = CreateSubmissionResolver.new(resolve_options)
```

Comment:
So this is 1:1 with `SubmissionService.create_submission`?

Reply:
The responsibilities of these Resolver classes are:

* gobble up arguments, context and object and then
* throw exceptions when things are not valid
* massage inputs and then send them to services and handle shaping the response

Does that answer your question?

Reply:
Makes sense!